### PR TITLE
release: v0.11.0

### PR DESCRIPTION
**Pull request by Geal**
# [0.11.0] - 2022-07-12

### Relax plugin api mutability ([PR #1340](https://github.com/apollographql/router/pull/1340) ([PR #1289](https://github.com/apollographql/router/pull/1289)

the `Plugin::*_service()` methods were taking a `&mut self` as argument, but since
they work like a tower Layer, they can use `&self` instead. This change
then allows us to move from Buffer to service factories for the query
planner, execution and subgraph services

By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1340 https://github.com/apollographql/router/pull/1289

### Add support to add custom resources on metrics. [PR #1354](https://github.com/apollographql/router/pull/1354)

Resources are almost like attributes but there are more globals. They are directly configured on the metrics exporter which means you'll always have these resources on each of your metrics. It could be pretty useful to set
a service name for example to let you more easily find metrics related to a specific service.

```yaml
telemetry:
  metrics:
    common:
      resources:
        # Set the service name to easily find metrics related to the apollo-router in your metrics dashboards
        service.name: "apollo-router"
```

By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1354

### Fix fragment on interface without typename [PR #1371](https://github.com/apollographql/router/pull/1371)

When the subgraph doesn't return the typename and the type condition of a fragment is an interface, we should return the values if the entity implements the interface

By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1371

### Fix detection of an introspection query [PR #1370](https://github.com/apollographql/router/pull/1370)

A query with at the root only one selection field equals to `__typename` must be considered as an introspection query

By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1370

### Accept nullable list as input [PR #1363](https://github.com/apollographql/router/pull/1363)

Do not throw a validation error when you give `null` for an input variable of type `[Int!]`.

By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1363

### Replace Buffers of tower services with service factories([PR #1289](https://github.com/apollographql/router/pull/1289) [PR #1355](https://github.com/apollographql/router/pull/1355))

tower services should be used by creating a new service instance for each new session
instead of going through a Buffer.

By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1289  https://github.com/apollographql/router/pull/1355

### execute the query plan's first response directly ([PR #1357](https://github.com/apollographql/router/issues/1357))

The query plan was entirely executed in a spawned task to prepare for the `@defer` implementation, but we can actually
generate the first response right inside the same future.

By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1357

### Remove deprecated `failure` crate from the dependency tree [PR #1373](https://github.com/apollographql/router/pull/1373)

This should fix automated reports about [GHSA-jq66-xh47-j9f3](https://github.com/advisories/GHSA-jq66-xh47-j9f3).

By [@yanns](https://github.com/yanns) in https://github.com/apollographql/router/pull/1373

### Various documentation edits ([PR #1329](https://github.com/apollographql/router/issues/1329))

By [@StephenBarlow](https://github.com/StephenBarlow) in https://github.com/apollographql/router/pull/1329

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,88 @@ All notable changes to Router will be documented in this file.
...
+## ❗ BREAKING ❗
+
+### Relax plugin api mutability ([PR #1340](https://github.com/apollographql/router/pull/1340) ([PR #1289](https://github.com/apollographql/router/pull/1289)
+
+the `Plugin::*_service()` methods were taking a `&mut self` as argument, but since
+they work like a tower Layer, they can use `&self` instead. This change
+then allows us to move from Buffer to service factories for the query
+planner, execution and subgraph services
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1340 https://github.com/apollographql/router/pull/1289
```
Comment (o0Ignition0o) [inline CHANGELOG.md:18]: we might wanna explain what this refactoring means for plugin writers, like they might have to rethink scopes (things that should be static might not be anymore)

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,88 @@ All notable changes to Router will be documented in this file.
...
+
+### Fix fragment on interface without typename [PR #1371](https://github.com/apollographql/router/pull/1371)
+
+When the subgraph doesn't return the typename and the type condition of a fragment is an interface, we should return the values if the entity implements the interface
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1371
+
+### Fix detection of an introspection query [PR #1370](https://github.com/apollographql/router/pull/1370)
+
+A query with at the root only one selection field equals to `__typename` must be considered as an introspection query
```
Comment (o0Ignition0o) [inline CHANGELOG.md:48]: A query that only contains `__typename` at the root must be considered as an introspection query

**Review by o0Ignition0o — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,8 +4,98 @@ All notable changes to Router will be documented in this file.
...
+into the new service in the `Plugin::*_service()` methods**.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1340 https://github.com/apollographql/router/pull/1289
+
+## 🚀 Features
+
+### Add support to add custom resources on metrics. [PR #1354](https://github.com/apollographql/router/pull/1354)
+
+Resources are almost like attributes but there are more globals. They are directly configured on the metrics exporter which means you'll always have these resources on each of your metrics. It could be pretty useful to set
+a service name for example to let you more easily find metrics related to a specific service.
```
Comment (abernix) [inline CHANGELOG.md:29]:
```suggestion
Resources are almost like attributes but more global. They are directly configured on the metrics exporter which means you'll always have these resources on each of your metrics.  This functionality can be used to, for example,
apply a `service.name` to metrics to make them easier to find in larger infrastructure, as demonstrated here:
```

Thoughts?

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,8 +4,98 @@ All notable changes to Router will be documented in this file.
...
+
+### Accept nullable list as input [PR #1363](https://github.com/apollographql/router/pull/1363)
+
+Do not throw a validation error when you give `null` for an input variable of type `[Int!]`.
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1363
+
+## 🛠 Maintenance
+
+### Replace Buffers of tower services with service factories([PR #1289](https://github.com/apollographql/router/pull/1289) [PR #1355](https://github.com/apollographql/router/pull/1355))
```
Comment (abernix) [inline CHANGELOG.md:64]:
```suggestion
### Replace Buffers of tower services with service factories ([PR #1289](https://github.com/apollographql/router/pull/1289) [PR #1355](https://github.com/apollographql/router/pull/1355))
```

formatting

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,8 +4,98 @@ All notable changes to Router will be documented in this file.
...
+
+tower services should be used by creating a new service instance for each new session
+instead of going through a Buffer.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1289  https://github.com/apollographql/router/pull/1355
+
+### execute the query plan's first response directly ([PR #1357](https://github.com/apollographql/router/issues/1357))
```
Comment (abernix) [inline CHANGELOG.md:71]:
```suggestion
### Execute the query plan's first response directly ([PR #1357](https://github.com/apollographql/router/issues/1357))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,8 +4,98 @@ All notable changes to Router will be documented in this file.
...
+
+The query plan was entirely executed in a spawned task to prepare for the `@defer` implementation, but we can actually
```
Comment (abernix) [inline CHANGELOG.md:73]:
```suggestion
The query plan was previously executed in a spawned task to prepare for the `@defer` implementation, but we can actually
```

**Review by abernix — APPROVED** [review]
Couple small things.  Real nice overall!

**Pull request merged**